ProfileManager.js

```diff
@@ -84,6 +92,118 @@ export default class ProfileManager {
     await this._sessionChanged({newData: session.data});
   }
 
+  // TODO: add docs
+  // FIXME: rename?
```

Comment:
Yeah, I think a rename might be needed given how much is going on in this function. This sounds more like it's some kind of profile initialization function.